src/main/java/org/mifos/connector/mpesa/camel/routes/PaybillRoute.java

```diff
@@ -27,80 +29,82 @@
 @Component
 public class PaybillRoute extends ErrorHandlerRouteBuilder {
     private final Logger logger = LoggerFactory.getLogger(this.getClass());
-    private final ObjectMapper objectMapper;
-    @Autowired
-    private ZeebeProcessStarter zeebeProcessStarter;
+    private ObjectMapper objectMapper = new ObjectMapper();
     @Autowired
     private ZeebeClient zeebeClient;
     @Value("${channel.host}")
     private String channelUrl;
+    @Value("${currency}")
+    private String currency;
+    @Value("${accountHoldingInstitutionId}")
+    private String accountHoldingInstitutionId;
+    @Autowired
+    private MpesaUtils mpesaUtils;
     private final String secondaryIdentifierName = "MSISDN";
-    @Value("${timer}")
-    private String timer;
-    private static HashMap<String, String> hm = new HashMap<>();
-
-    public PaybillRoute(ObjectMapper objectMapper) {
-        this.objectMapper = objectMapper;
-    }
+    //    private RedisTemplate<String, String> redisTemplate;
+    @Autowired
+    private MpesaPaybillProp mpesaPaybillProp;
 
     @Override
     public void configure() {
+
         from("rest:POST:/validation")
                 .id("mpesa-validation")
                 .unmarshal().json(JsonLibrary.Jackson, PaybillRequestDTO.class)
                 .log(LoggingLevel.INFO, "## Paybill request payload")
                 .setBody(e -> {
                     PaybillRequestDTO paybillRequestDTO = e.getIn().getBody(PaybillRequestDTO.class);
                     //Getting the ams name
-                    String amsName = e.getIn().getHeader("amsName", String.class);
-                    String dfspId = e.getIn().getHeader("dfspId", String.class);
-                    String currency = e.getIn().getHeader("currency", String.class);
-                    //Channel URL
-                    logger.debug("Channel URL : {}", channelUrl);
+                    String businessShortCode = paybillRequestDTO.getShortCode();
+                    String amsName = mpesaPaybillProp.getAMSFromShortCode(businessShortCode);
+                    String amsUrl = mpesaUtils.getAMSUrl(amsName);
+
+                    e.getIn().setHeader("amsUrl", amsUrl);
+                    e.getIn().setHeader("amsName", amsName);
+                    e.getIn().setHeader("accountHoldingInstitutionId", accountHoldingInstitutionId);
                     e.setProperty("channelUrl", channelUrl);
                     e.setProperty("secondaryIdentifier", secondaryIdentifierName);
                     e.setProperty("secondaryIdentifierValue", paybillRequestDTO.getMsisdn());
-                    e.setProperty("dfspId", dfspId);
                     ChannelRequestDTO obj = MpesaUtils.convertPaybillPayloadToChannelPayload(paybillRequestDTO, amsName, currency);
-                    return obj.toString();
+                    String channelRequestDTO = null;
```

Comment:
You don't need this variable. You can return Directly from the try block and  throw custom exception in catch block

Reply:
Done